src/lib/Scenes/City/Components/AllEvents.tsx

```diff
@@ -28,11 +29,26 @@ export class AllEvents extends React.Component<Props, State> {
   }
 
   componentDidMount() {
-    this.updateSections()
+    this.updateSections(this.props)
   }
 
-  updateSections = () => {
-    const { buckets, cityName } = this.props
+  componentWillReceiveProps(nextProps) {
+    const shouldUpdate = ["saved", "closing", "museums", "opening", "closing"]
+      .map(key => {
+        return !isEqual(
+          this.props.buckets[key].map(g => g.node.is_followed),
+          nextProps.buckets[key].map(g => g.is_followed)
+        )
+      })
+      .some(a => a)
```

Comment:
This is not important at all, but as an FYI—replacing the `map` call with `some` would save on extra comparisons after a bucket did not pass the comparator.

Reply:
Oh that's true, even more efficient!